x-pack/plugins/maps/public/components/metrics_editor/metric_editor.tsx

```diff
@@ -12,11 +12,17 @@ import { EuiButtonEmpty, EuiComboBoxOptionOption, EuiFieldText, EuiFormRow } fro
 import { FormattedMessage } from '@kbn/i18n/react';
 import { MetricSelect } from './metric_select';
 import { SingleFieldSelect } from '../single_field_select';
-import { AggDescriptor } from '../../../common/descriptor_types';
+import { AggDescriptor, FieldedAggDescriptor } from '../../../common/descriptor_types';
 import { AGG_TYPE } from '../../../common/constants';
 import { getTermsFields } from '../../index_pattern_util';
 import { IFieldType } from '../../../../../../src/plugins/data/public';
 
+export function isMetricInvalid(aggDescriptor: AggDescriptor) {
```

Comment:
a nit. but would it be more readable as `isMetricValid` and avoid the compsci-101 trippy conditional?